src/Tokenizer/Tokens.php

```diff
@@ -391,8 +391,9 @@ public function clearChanged(): void
     public function clearEmptyTokens(): void
     {
         $limit = \count($this);
+        $index = 0;
 
-        for ($index = 0; $index < $limit; ++$index) {
+        for (; $index < $limit; ++$index) {
```

Comment:
FYI @keradus I have most of the PHPStan issues fixed locally, but I did not create PR yet because I wanted to clear the current level, and I did not have time to finish it.

Reply:
@keradus in general, it's related to [`polluteScopeWithLoopInitialAssignments`](https://phpstan.org/config-reference#pollutescopewithloopinitialassignments) in PHPStan. Strict rules plugin sets it to false, so loops are treated like separate scopes. The solution for this in our case is to extract initial assignments before the loop, but I don't like it and I would prefer to enable this option again (and that's what I did in my branch).

Please revert these loop fixes as I started to work on PHPStan fixes ~2 weeks ago and have more of them. I will create PR soon, 24 errors left 😅.

Reply:
5ce383aaba1d4a57b84d219f8365717591f7750c reverted